src/Symfony/Component/Security/Http/Firewall/LogoutListener.php

```diff
@@ -90,16 +118,12 @@ public function authenticate(RequestEvent $event)
             }
         }
 
-        $response = $this->successHandler->onLogoutSuccess($request);
-        if (!$response instanceof Response) {
-            throw new \RuntimeException('Logout Success Handler did not return a Response.');
-        }
+        $logoutEvent = new LogoutEvent($request, $this->tokenStorage->getToken());
+        $this->eventDispatcher->dispatch($logoutEvent);
 
-        // handle multiple logout attempts gracefully
-        if ($token = $this->tokenStorage->getToken()) {
-            foreach ($this->handlers as $handler) {
-                $handler->logout($request, $response, $token);
```

Comment:
@wouterj 

I'm currently working on the FosHttpCacheBundle which has a [LogoutHandler](https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/396eea2c79661fe591a80788de4d7ae671c76b49/src/Security/Http/Logout/ContextInvalidationLogoutHandler.php#L26) but since Symfony 5.1 this seems not longer be called in the [test](https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/396eea2c79661fe591a80788de4d7ae671c76b49/tests/Functional/Security/Http/Logout/ContextInvalidationLogoutHandlerTest.php#L47):
https://travis-ci.org/github/FriendsOfSymfony/FOSHttpCacheBundle/jobs/714724517

For me it looks like it has todo with this change, but I'm not 100% sure. For the Bundle its not important as that service is deprecated and the thing handled by another listener but maybe it could be important for other project when the LogoutHandler is not called.

Reply:
That's because it's registered in a [very weird way](https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/blob/9fd07c2256e8974f25a1da2a0c8e616cab3881f3/src/DependencyInjection/FOSHttpCacheExtension.php#L315):

```php
$container->setAlias('security.logout.handler.session', 'fos_http_cache.user_context.session_logout_handler');
```

All internal handlers are now logout listeners, so "overriding" a core service ID no longer works. I'm not sure if internal core service IDs are covered by the BC promise (cc @nicolas-grekas please confirm).